### PR TITLE
Make attributes of Site Api configurable

### PR DESCRIPTION
**Pull request by tlueder**
Fixes https://github.com/sascha-egerer/phpstan-typo3/issues/52

### CODE REVIEW

File: src/Type/SiteDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	{
+		$argument = $methodCall->getArgs()[0] ?? null;
+
+		if ($argument === null || !($argument->value instanceof \PhpParser\Node\Scalar\String_)) {
+			return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+		}
+
+		if (isset($this->siteApiGetAttributeMapping[$argument->value->value])) {
+
+			$phpType = TypeNamingUtility::translateTypeNameToType($this->siteApiGetAttributeMapping[$argument->value->value]);
```
Comment (sascha-egerer) [inline src/Type/SiteDynamicReturnTypeExtension.php:46]: Can we use \PHPStan\PhpDoc\TypeNodeResolver instead?

Comment (tlueder) [reply src/Type/SiteDynamicReturnTypeExtension.php:46]: I already did that. :)

BTW should I change the name from siteApiGetAttributeMapping to siteGetAttributeMapping as well to match with requestGetAttributeMapping from your PR?

Comment (sascha-egerer) [reply src/Type/SiteDynamicReturnTypeExtension.php:46]: Yeah, sounds good. The "Api" thing makes sense for "Context API" but IMO not for the other stuff

**Comment by sascha-egerer** [comment]
Looks good but some tests should be added.

**Comment by tlueder** [comment]
I'll do that once https://github.com/sascha-egerer/phpstan-typo3/pull/60 is merged. And steal your work..

**Comment by tlueder** [comment]
@sascha-egerer any idea what I do wrong here? Only the test for the custom attribute is failing but I don't know why.

**Pull request merged**